web3/middleware/cache.py

```diff
@@ -267,19 +268,20 @@ def middleware(method, params):
 AVG_BLOCK_TIME_UPDATED_AT_KEY = 'avg_block_time_updated_at'
 
 
-def _should_cache_by_latest_block(method, params, response=None):
-    if method == 'eth_getBlockByNumber':
-        if params == ['latest'] or params == ['pending']:
-            return False
-
-    return _should_cache(method, params, response)
+def _is_latest_or_pending_block_request(method, params):
+    if method != 'eth_getBlockByNumber':
+        return False
+    elif params == ['latest']:
+        return True
```

Comment:
It's called `is_latest_or_pending` but caches only `latest`?

Reply:
Last minute change that didn't make it into the rename of the function.  fixing.